modules/microshift-custom-ca-proc.adoc

```diff
@@ -0,0 +1,143 @@
+
+. For each custom CA that you need, add an `apiServer` section called `namedCertificates` to the `/etc/microshift/config.yaml` {microshift-short} configuration file by using the following example:
++
+[source,yaml]
+----
+apiServer:
+  namedCertificates:
+   - certPath: ~/certs/api_fqdn_1.crt <1>
+     keyPath:  ~/certs/api_fqdn_1.key <2>
+   - certPath: ~/certs/api_fqdn_2.crt
+     keyPath:  ~/certs/api_fqdn_2.key
+     names: <3>
+     - api_fqdn_1
+     - *.apps.external.com
+----
+<1> Add the full path to the certificate.
+<2> Add the full path to the certificate key.
+<3> Optional. Add a list of explicit DNS names. Leading wildcards are allowed. If no names are provided, the implicit names are extracted from the certificates.
+
+. Restart the {microshift-service} to apply the certificates by running the following command:
++
+[source,terminal]
+----
+$ systemctl microshift restart
+----
+
+. Wait a few minutes for the system to restart and apply the custom server. New `kubeconfig` files are generated in the `/var/lib/microshift/resources/kubeadmin/` directory.
+
+. Copy the `kubeconfig` files to the client. If you specified wildcards for the IP address, update the `kubeconfig` to remove the public IP address of the server and replace that IP address with the specific wildcard range you want to use.
+
+. From the client, use the following steps:
+
+.. Specify the `kubeconfig` to use by running the following command:
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/custom-kubeconfigs/kubeconfig <1>
+----
+<1> Use the location of the copied `kubeconfig` file as the path.
+
+.. Check that the certificates are applied by using the following command:
++
+[source,terminal]
+----
+$ oc --certificate-authority ~/certs/ca.ca get node
+----
++
+.Example output
+[source,terminal]
+----
+oc get node
+NAME                             STATUS   ROLES                         AGE   VERSION
+dhcp-1-235-195.arm.example.com   Ready    control-plane,master,worker   76m   v1.29.2
+----
+
+.. Add the new CA file to the $KUBECONFIG environment variable by running the following command:
++
+[source,terminal]
+----
+$ oc config set clusters.microshift.certificate-authority /tmp/certificate-authority-data-new.crt
+----
+
+.. Verify that the new certificate was generated by running the following command:
++
+[source,terminal]
+----
+$ oc config view --flatten
+----
++
+.Example externally generated `kubeconfig` file
++
+[source,yaml]
+----
+apiVersion: v1
+clusters:
+- cluster:
+    server: https://api.ci-ln-k0gim2b-76ef8.aws-2.ci.openshift.org:6443
+  name: ci-ln-k0gim2b-76ef8
+contexts:
+- context:
+    cluster: ci-ln-k0gim2b-76ef8
+    user:
+  name:
+current-context:
+kind: Config
+preferences: {}
+----
++
+[NOTE]
+====
+The `certificate-authority-data` section is not present in externally generated `kubeconfig` files.
```

Comment:
this warning is no longer valid ,since we added the `certificate-authority-data`  in the previous step, please remove it

Reply:
it is not in the example file that preceeds the Note